MonMaraicher/Features/Map/MapView.swift

```diff
@@ -14,23 +14,34 @@ struct MapView: View {
 
     var body: some View {
         Map(position: $viewModel.mapCameraPosition, selection: $viewModel.selectedMarker) {
-
             ForEach(viewModel.allMarkers, id: \.id) { marker in
                 Marker(marker.title, systemImage: marker.systemImage, coordinate: marker.coordinate)
                     .tag(marker)
                     .tint(.orange)
             }
-
             UserAnnotation()
-
         }
         .mapStyle(.standard(elevation: .realistic))
         .mapControls {
             MapUserLocationButton()
         }
+        // FIXME: - method to rework
+        .onChange(of: viewModel.hasUserAcceptedLocation ) {
+            viewModel.onReloadingFarmersButtonTapped()
+        }
         .onAppear(perform: viewModel.onViewAppear)
         .overlay(alignment: .bottom) {
-            nearbyFarmerButton
+            HStack {
+                nearbyFarmerButton
+                reloadFarmersButton
+                    .disabled(viewModel.farmersLoadingInProgress)
+            }
+        }
+        .overlay {
+            if viewModel.farmersLoadingInProgress && viewModel.hasUserAcceptedLocation {
```

Comment:
I would create a computed property called `isProgressShowed` in your viewModel returning the result of `viewModel.farmersLoadingInProgress && viewModel.hasUserAcceptedLocation`

Reply:
But it actually more than that since the progress view is hidding your whole view. So it more a view state. We should discuss about that.